### PR TITLE
ENH: Add three new parameters to run_amrfinderplus_n

### PR DESCRIPTION
**Pull request by VinzentRisch**
solves #96 

Adds three new parameters to run_amrfinderplus_n. Two of those can be used with the action annotate_sample_data_amrfinderplus.

### Set up an environment 
https://github.com/bokulich-lab/q2-amr?tab=readme-ov-file#installation

### Run it locally 
1. First, clone the repo and checkout the PR branch:
```bash
git clone https://github.com/bokulich-lab/q2-amr.git
cd q2-amr
git fetch origin pull/97/head:pr-97
git checkout pr-97
pip install -e .
```
2. Download test files 
[PR-88.zip](https://github.com/user-attachments/files/16250258/PR-88.zip)

3. Test it out!
```bash
qiime amr fetch-amrfinderplus-db --o-amrfinderplus-db amrfinderplus_db.qza 
qiime amr annotate-sample-data-amrfinderplus --i-sequences test_mags.qza --output-dir output_mags --i-amrfinderplus-db amrfinderplus_db.qza --p-plus --p-curated-ident --p-organism "Escherichia_coli_Shigella" --p-report-all-equal --p-coverage-min 0.4 --p-translation-table "11" --p-gpipe-org True --p-report-common
qiime amr annotate-sample-data-amrfinderplus --i-sequences test_contigs.qza --output-dir output_contigs --i-amrfinderplus-db amrfinderplus_db.qza --p-plus --p-curated-ident --p-organism "Escherichia_coli_Shigella" --p-report-all-equal --p-coverage-min 0.4 --p-translation-table "11" --p-gpipe-org True --p-report-common
```

### CODE REVIEW

**Comment by VinzentRisch** [comment]
@misialq could you have a look at the added typemap in this PR.
so i want to have different choices for one parameter depending if one parameter is true or false 
And i have this. I saw that placeholder type with _ and Int in moshpit, but there it also says it is using Int because visualizations dont have types so Im not sure if i should be doing it like this.
```
P_gpipe_org, P_organism, _ = TypeMap(
    {
        (Bool % Choices(True), Str % Choices(organisms_gpipe)): Int,
        (Bool % Choices(False), Str % Choices(organisms)): Int,
    }
)
```
Also this leads to the parameter gpipe_org not behaving like a bool parameter anymore. So it is not possible to just put --p-gpipe-org but you have to give it a value of eighter True or False. Is this the normal behaviour because I use Bool % Choices(True)? Or is there a better solution?

**Comment by misialq** [comment]
Hey @VinzentRisch, I think it's fine for the Int as a placeholder - you are not using it after all. For the boolean, though, I'm not sure if there is a better way - we have it like this somewhere in MOSHPIT and Santiago spent some time exploring this (ending up with a solution like yours) but maybe we could ask tomorrow just to be sure?

File: q2_amr/plugin_setup.py [hunk 1]
```diff
@@ -1204,6 +1242,9 @@
         "coverage_min": "Minimum proportion of reference gene covered for a "
         "BLAST-based hit (Methods BLAST or PARTIAL).",
         "translation_table": "Translation table used for BLASTX.",
+        "report_common": "Report proteins common to a taxonomy group.",
+        "gpipe_org": "Use Pathogen Detection taxgroup names as arguments to the "
+        "organism option",
```
Comment (ChristosMatzoros) [inline q2_amr/plugin_setup.py:1247]:
```suggestion
        "organism option.",
```

**Review by ChristosMatzoros — CHANGES_REQUESTED** [review]
Hi @VinzentRisch,

Runs as expected with no errors and code looks good. See a minor change to make and after that we are set to merge.

**Comment by VinzentRisch** [comment]
Thanks for the review, added the full stop :)

**Review by ChristosMatzoros — APPROVED** [review]

**Pull request merged**